### DISCUSSION TITLE
Server didn't start after latest update, decided to restore and now getting different errors.

### DISCUSSION THREAD

**Discussion by trop1kal** [immich-app/immich#14664] [Q&A]
When restoring using the instructions via the link below I'm getting multiple invalid command errors.
https://immich.app/docs/administration/backup-and-restore/#manual-backup-and-restore

Any suggestions on why I'm getting these errors when completing the restore?
Followed the commands..

docker compose down -v  # CAUTION! Deletes all Immich data to start from scratch
## Uncomment the next line and replace DB_DATA_LOCATION with your Postgres path to permanently reset the Postgres database
Remove-Item -Recurse -Force D:\immich-app\postgres # CAUTION! Deletes all Immich data to start from scratch
docker compose pull             # Update to latest version of Immich (if desired)
docker compose create           # Create Docker containers for Immich apps without running them
docker start immich_postgres    # Start Postgres server
sleep 10                        # Wait for Postgres server to start up
# Check the database user if you deviated from the default
gc "D:\immich-app\library\backups\immich-db-backup-1732932000008.sql.gz" | docker exec -i immich_postgres psql --username=postgres  # Restore Backup
docker compose up -d            # Start remainder of Immich apps

invalid command \j?{J????uo_???<i??????????{▼?pt,??n?[@↑6????♥A?????U???y)?K!?i??J??????eum
invalid command \?X↕?p??6????????►??▬u♂ZZ?g{gC???q
invalid command \8m{☺#"↓??x???'↕??,???k?=??=????????~Y??_????????∟7??kj?↕??<?15??
invalid command \♫0?W?3??"?♣??Vp??[??}?@????pV)?↓U?bA??????‼?&IJ??j???=$???pD6?0?>??↑?◄?_S?H♫???☻y???_??U%????-??↨????m@?l??M?L???;v♦??x?§▬?<??▼
invalid command \?/?{????49Q???~⌂??`☺                                                             ???♫♂D??¶?H???6▬?fx??♣
invalid command \{z►?|A????,?§??C???▼g??????→?Xv??}9?D??♥??ka☼G♣↔?z??↑yYb?????↨?↨6?6???Xf???)§??????3j?↔c]V??|?☼Vl??"??→?????v?☺?>$F??e??T¶B2?Ya5??????♣????>LD☺???????????♥??$J??}?↨T??????op♥?O???HO??????N??r?+??j???{|&I????6??♫?
invalid command \??FYD??
invalid command \N☻?T?p☼?????C???V&?.?x'♦)Q.V?9??3&&???F???_y???!??????v)*?A??►p??R♂?.?????{??♂⌂?!:????H????aZ??☼?????6&]?'?▲☺??∟→?☺????%???8§1?H?X?u}n?▼???^"?♠j?I?%∟♥??/0§?!??g→?B???Y∟??L??O▬☼??/??ly??0??W♥Z??B*?z♥¶§'>????D?FE??'
invalid command \?♂?cIP7▬Z?uB⌂?
invalid command \???j??2????3?►?
invalid command \??T?????<?cALJ??♠??0??↨▲?>~n??g♫.?r?↓??↑??◄????↔???QR=?g♂^{?♫???∟☺u?0???(??@t?????
invalid command \¶.?&?#?I??#?>?@???►?p?q_☺????‼8???????
invalid command \S???Y?-??lO??+?c?/??$F↑~?%x?Kw:????8D?M↓
invalid command \♣??|????+♦????`?▲?}???♠→?↑??????↨5+N?O?▲????????y?V?▬9??Z?I??"j?????f??∟♣?`??☻??§6↓?
invalid command \?PU?§??_?‼?*??☼B???J????P☼??b?S?L??p
invalid command \???L??????∟?_r????,?n?Y??j?►?9?t?§?u??E??=‼0?[z
invalid command \?♫h?X"`???b1♂]??F[??uEa??u???r?cT$3?????y↕H??♥:??p8V??.??¶?b???l?▬J?
invalid command \?Z?
invalid command \??♥
invalid command \?t?8??u?K??h?????'▲j?O♂???*???
invalid command \???j6Z???`????)?=????↨☺?>???▼
invalid command \Vc*?9h?☺h?l????????_⌂??s??????=~???~????????⌂??????D/♫y???*???????K?Rk?⌂?9?M??'???m,??
invalid command \?a?u?▲Z??1????||TX??e@?a6?☻‼??h◄????.◄??|??'????C?↑?x?D↑?N??^.me??y??&??Q??-☻lQ??S§(↑|d_9"☼K?↔????????
invalid command \?z6?↕?$?'???f►b?z?jV3C???J???<r▬w?↨??O:/+?v?#??F~????2???????~6??v8♥?0??n?
invalid command \????F↨↓?►???↓?
invalid command \f??*6US????!
invalid command \R?↨???,?
invalid command \?☺|?5?Eg∟?9???wS??↓L?K~☼♦?2i?▼???→B}-☺I?%j?P???&↕#?o:?:FR‼?:►0+?6=A
invalid command \S?n?8_%↔??#/~?n?☼N?R?♂??◄?5I???▲??U??'?X:???????B☺?]♠¶????V8?=?oW#)oK↑{f???

**Comment by mmomjian** [comment]
You haven’t shared any information at all about what commands you ran or what you tried. That in mind, in guessing you’re trying to import a file without un-GZIPing

**Reply by trop1kal** [reply]
Thanks for the quick response! Much appreciated.

I ran the following to restore..

docker compose down -v  # CAUTION! Deletes all Immich data to start from scratch
## Uncomment the next line and replace DB_DATA_LOCATION with your Postgres path to permanently reset the Postgres database
Remove-Item -Recurse -Force D:\immich-app\postgres # CAUTION! Deletes all Immich data to start from scratch
docker compose pull             # Update to latest version of Immich (if desired)
docker compose create           # Create Docker containers for Immich apps without running them
docker start immich_postgres    # Start Postgres server
sleep 10                        # Wait for Postgres server to start up
# Check the database user if you deviated from the default
gc "D:\immich-app\library\backups\immich-db-backup-1732932000008.sql.gz" | docker exec -i immich_postgres psql --username=postgres  # Restore Backup
docker compose up -d            # Start remainder of Immich apps

**Reply by mmomjian** [reply]
Your backup is a GZ file. You should mount it into the container and restore using the instructions in the Linux section, using `gunzip`. You can make a ticket on discord if you need more help.

**Comment by trop1kal** [comment]
I tried unpacking the file and restoring via command

gc "D:\immich-app\library\backups\immich-db-backup-1732932000008.sql" | docker exec -i immich_postgres psql --username=postgres # Restore Backup

I get the following error.. 
The sql file is 2.25gb in size.. Do I have to increase the size of immich_postgres some how?



Program 'docker.exe' failed to run: capacity was less than the current size.
Parameter name: requiredLengthAt line:1 char:73
+ ... 0005.sql" | docker exec -i immich_postgres psql --username=postgres   ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~.
At line:1 char:1
+ gc "D:\immich-app\library\backups\immich-db-backup-1733536800005.sql" ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (:) [], ApplicationFailedException
    + FullyQualifiedErrorId : NativeCommandFailed

**Reply by mmomjian** [reply]
You will need to mount the file into the container and run the restore in a docker shell.

**Reply by tkp-dev** [reply]
Hi. Have the same problem after latest Windows updates restart today with Immich v1.122.2 on Docker/Windows.

After postgres failed to start
```
[28] FATAL:  could not create file "base/16384/812865": File exists
```
decided to restore from backup using Windows workflow from here https://immich.app/docs/administration/backup-and-restore#manual-backup-and-restore :

```
docker compose down -v
Remove-Item -Recurse -Force "C:\immich\postgres"
docker compose pull
docker compose create
docker start immich_postgres
sleep 20
gc "S:\immich\library\backups\1.sql" | docker exec -i immich_postgres psql --username=postgres
docker compose up -d
```

In powershell getting error for "gc "S:\immich\library\backups\1.sql" | docker exec -i immich_postgres psql --username=postgres":

```
Program 'docker.exe' failed to run: capacity was less than the current size.
Parameter name: requiredLengthAt line:9 char:40
+ ... ps\1.sql" | docker exec -i immich_postgres psql --username=postgres   ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~.
At line:9 char:1
+ gc "S:\immich\library\backups\1.sql" | docker exec -i immich_postgres ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (:) [], ApplicationFailedException
    + FullyQualifiedErrorId : NativeCommandFailed
```

Unpacked backup dump.sql size 5.27GB. Obviously for something in the Windows/Docker it is too large. Are there any suggestions what to change in a Windows environment to restore?

@mmomjian would you be so kind to provide a bit more details on your proposal to mount (how?) the backup to container and how to restore. Should one use restore command from here for Linux system https://immich.app/docs/administration/backup-and-restore#manual-backup-and-restore or other one? Docker shell - is it Powershell Terminal inside Docker or is it Docker/Containers/immich_postgres/Exec? Thank you!

**Reply by mmomjian** [reply]
You can look at the preview for the next docs version: https://main.preview.immich.app/docs/administration/backup-and-restore

you should mount the backup into the container, then restore from there

**Reply by tkp-dev** [reply]
Looks like this worked out. Have used Windows system (PowerShell). Thank you, 

How to mount backup explained inside the powershell code:

```
## You should mount the backup (as a volume, example: - 'C:\path\to\backup\dump.sql':/dump.sql) into the immich_postgres container using the docker-compose.yml
```
